packages/eas-cli/src/commands/branch/view.ts

```diff
@@ -25,82 +18,33 @@ export default class BranchView extends EasCommand {
   ];
 
   static flags = {
-    json: Flags.boolean({
-      description: `return a json with the branch's ID name and recent update groups.`,
-      default: false,
-    }),
+    'non-interactive': EasFlags['non-interactive'],
+    json: EasFlags.json,
+    limit: EasFlags.limit,
+    offset: EasFlags.offset,
   };
 
   async runAsync(): Promise<void> {
     let {
-      args: { name },
-      flags: { json: jsonFlag },
+      args: { name: branchName },
+      flags,
     } = await this.parse(BranchView);
-    if (jsonFlag) {
+    if (flags.json) {
       enableJsonOutput();
     }
 
     const projectDir = await findProjectRootAsync();
     const exp = getExpoConfig(projectDir);
     const projectId = await getProjectIdAsync(exp);
 
-    if (!name) {
-      const validationMessage = 'Branch name may not be empty.';
-      if (jsonFlag) {
-        throw new Error(validationMessage);
-      }
-      ({ name } = await promptAsync({
-        type: 'text',
-        name: 'name',
-        message: 'Please enter the name of the branch to view:',
-        validate: value => (value ? true : validationMessage),
-      }));
-    }
-
-    const { app } = await UpdateQuery.viewBranchAsync({
-      appId: projectId,
-      name,
-    });
-    const UpdateBranch = app?.byId.updateBranchByName;
-    if (!UpdateBranch) {
-      throw new Error(`Could not find branch "${name}"`);
+    if (!branchName) {
+      ({ name: branchName } = await selectBranchForProjectAsync(
+        projectId,
+        'Which branch would you like to view?',
+        { ...flags, offset: 0, limit: BRANCHES_LIMIT }
```

Comment:
you are misinterpreting this line-- 
because it occurring inside of a `if (!branchName) ...` block, it's prompting the user to select a branch to view, instead of failing outright as the command used to do. In this case we want to ignore the offset and limit provided by the user as input to the command, because it was never intended to influence this query.

Reply:
changed this line a bit, hope this helps